src/v/redpanda/admin_server.cc

```diff
@@ -1729,12 +1767,27 @@ admin_server::delete_user_handler(std::unique_ptr<ss::httpd::request> req) {
 
 ss::future<ss::json::json_return_type>
 admin_server::update_user_handler(std::unique_ptr<ss::httpd::request> req) {
+    if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+        // In order that we can do a reliably ordered validation of
+        // the request (and drop no-op requests), run on controller leader;
+        throw co_await redirect_to_leader(*req, model::controller_ntp);
```

Comment:
What was the behavior before? Did we just forward the request to the leader internally?

Reply:
We would throw a not_leader error from security_frontend and that would turn into a redirect via `throw_on_error`.  That happens after we've already written to the raft log, so not helpful for any checks that happen before that.